app/graphql/resolvers/surveys_search.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "search_object/plugin/graphql"
+require "graphql/query_resolver"
+
+class Resolvers::SurveysSearch
+  include SearchObject.module(:graphql)
+
+  # NOTE: following line is needed by the SearchObject module but cannot be used in `fetch_results`
+  scope { Survey::Poll.filtered_for_current_user(context[:current_user]) }
+
+  type types[Types::QueryTypes::SurveyPollType]
+
+  option :ids, type: types[types.ID]
+
+  # https://github.com/nettofarah/graphql-query-resolver
+
+  def fetch_results
+    # NOTE: scope from class is not available here, so we need to create a local scope here :/
+    scope = Survey::Poll.filtered_for_current_user(context[:current_user])
+
+    # select only available ids
+    scope = scope.where(id: ids) if ids.present? && ids.any?
```

Comment:
seit wann schreiben wir das hier unten in fetch results rein und nicht mehr als eigenen ausgelagerte Methoden?

Reply:
als ausgelagerte methode ging der name `scope` nicht

Reply:
wenn wir auf filter umstellen hat es den vorteil, dass es den workaround für scope nicht braucht.